File: caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    bool error = 1;
    if (argc == 2)
    {
        int key = atoi(argv[1]);
        if (key)
        {
            int num_plus;
            string text = get_string("plaintext: ");
            for (int i = 0, n = strlen(text); i < n; i++)
            {
                num_plus = 0;
                if (islower(text[i]))
                {
                    num_plus = 97;
                }
                else if (isupper(text[i]))
                {
                    num_plus = 65;
                }
                if (num_plus)
                {
                    text[i] = (text[i] + key - num_plus) % 26 + num_plus;
                }
            }
            printf("ciphertext: %s\n", text);
            return 0;
        }
    }
    
    if (error == 1)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
}

File: caesar.py
from cs50 import get_string
from sys import argv

if len(argv) != 2:
    exit("Usage: python caesar.py k")
plaintext = get_string("plaintext:  ")
cypher = ""
k = int(argv[1])
for c in plaintext:
    if c.isalpha():
        if c.isupper():
            plus = 65
        else:
            plus = 97
        new_c = c
        new_c = ord(new_c)
        new_c = chr((new_c - plus + k) % 26 + plus)
    else:
        new_c = c
    print(new_c)
    cypher = cypher + new_c
print(f"ciphertext: {cypher}")